### ISSUE TITLE
model.fit_transform() bug?

### ISSUE THREAD

**Issue by xiankgx** [MaartenGr/BERTopic#103]
```
topic_model = BERTopic(nr_topics=30)
topics, _ = topic_model.fit_transform(docs)
topics1, _ = topic_model.transform(docs)

topics1 == topics  # returns False
```

```
topic_model = BERTopic(nr_topics=30)
topic_model.fit(docs)
topics, _ = topic_model.transform(docs)
topics1, _ = topic_model.transform(docs)

topics1 == topics  # returns True
```

I was expecting topics == topics1 to be equal in the first code.

**Comment by xiankgx** [comment]
bertopic==0.6.0
numpy==1.20.2
torch==1.8.0
torchaudio==0.8.0a0+a751e1d
torchserve==0.3.0
torchvision==0.9.0

Here are the libraries

**Comment by MaartenGr** [comment]
Thank you for the nice example so it can easily be reproduced. I tested out some stuff quickly and it seems one of two things. Either the mapping of topics after reducing them has some issues or it is an issue with HDBSCAN (see [this](https://github.com/scikit-learn-contrib/hdbscan/issues/445) and [this](https://github.com/scikit-learn-contrib/hdbscan/issues/361)). 

As it turns out, there might be a very small chance HDBSCAN predicts unforeseen labels with `approximate_predict`. It is a function that allows you to also extract the probabilities of each point belonging to a specific cluster. I have had issues with this function in the past as it can slow down BERTopic significantly. Seeing as it only affects a very small percentage, I would continue either function until it gets fixed.

**Comment by MaartenGr** [comment]
@xiankgx BERTopic v0.8 was just released and I believe it fixes the issue you were encountering. I will close this for now but if you still encounter this issue feel free to reopen it!
